logging/src/Logging4cplus.cpp

```diff
@@ -0,0 +1,53 @@
+// Concord
+//
+// Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license,
+// as noted in the LICENSE file.
+
+#include "Logger.hpp"
+#include <fstream>
+#include <iostream>
+
+#include <log4cplus/logger.h>
+#include <log4cplus/configurator.h>
+#include <log4cplus/helpers/property.h>
+#include <log4cplus/consoleappender.h>
+#include <log4cplus/fileappender.h>
+
+using namespace log4cplus;
+
+namespace logging {
+
+const char* logPattern = "%X{rid}|%d{%m-%d-%Y %H:%M:%S.%q}|%-5p|%c|%X{thread}|%M|%m%n";
+
+void initLogger(const std::string& configFileName) {
+  std::ifstream infile(configFileName);
```

Comment:
You may need an Initializer here: https://sourceforge.net/p/log4cplus/wiki/CodeExamples/
However, possibly not. If tests all pass and system works, no worries :)

Reply:
Fixed with force initialization to avoid log4cplus warnings in tests

Reply:
> Fixed with force initialization to avoid log4cplus warnings in tests

What is forced initialization? You're doing too advanced stuff for me haha :)

Reply:
Previously I did:
`logging::Logger GL = defaultInit()`
But if in a particular test GL was not used, defultInit() hasn't happened and we got this ugly
`log4cplus:ERROR No appenders could be found for logger (thread-pool).
log4cplus:ERROR Please initialize the log4cplus system properly.
`
Now I do:
`Logger getLogger(const std::string& name) {
  static bool __logging_init__ = defaultInit();  // one time initialization
  (void)__logging_init__;
  return log4cplus::Logger::getInstance(name);
}
`
and defaultInit happens once before _any_ logger is used.

Reply:
Thanks for the explanation.